cmd/http.go

```diff
@@ -0,0 +1,168 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Edited by the systemk authors in 2021.
+
+package cmd
+
+import (
+	"context"
+	"crypto/tls"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/virtual-kubelet/systemk/internal/provider"
+	"github.com/virtual-kubelet/virtual-kubelet/log"
+	"github.com/virtual-kubelet/virtual-kubelet/node/api"
+)
+
+// AcceptedCiphers is the list of accepted TLS ciphers, with known weak ciphers elided
+// Note this list should be a moving target.
+var AcceptedCiphers = []uint16{
+	tls.TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,
+	tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,
+	tls.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,
+	tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
+
+	tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
+	tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+	tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+	tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+}
+
+func loadTLSConfig(certPath, keyPath string) (*tls.Config, error) {
+	cert, err := tls.LoadX509KeyPair(certPath, keyPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "error loading tls certs")
+	}
+
+	return &tls.Config{
+		Certificates:             []tls.Certificate{cert},
+		MinVersion:               tls.VersionTLS12,
+		PreferServerCipherSuites: true,
+		CipherSuites:             AcceptedCiphers,
+	}, nil
+}
+
+func setupHTTPServer(ctx context.Context, p provider.Provider, cfg *apiServerConfig, getPodsFromKubernetes api.PodListerFunc) (_ func(), retErr error) {
+	var closers []io.Closer
+	cancel := func() {
+		for _, c := range closers {
+			c.Close()
+		}
+	}
+	defer func() {
+		if retErr != nil {
+			cancel()
+		}
+	}()
+
+	if cfg.CertPath == "" || cfg.KeyPath == "" {
+		log.G(ctx).
+			WithField("certPath", cfg.CertPath).
+			WithField("keyPath", cfg.KeyPath).
+			Error("TLS certificates not provided, not setting up pod http server")
+	} else {
+		tlsCfg, err := loadTLSConfig(cfg.CertPath, cfg.KeyPath)
+		if err != nil {
+			return nil, err
+		}
+		l, err := tls.Listen("tcp", cfg.Addr, tlsCfg)
+		if err != nil {
+			return nil, errors.Wrap(err, "error setting up listener for pod http server")
+		}
+
+		mux := http.NewServeMux()
+
+		podRoutes := api.PodHandlerConfig{
+			RunInContainer:        p.RunInContainer,
+			GetContainerLogs:      p.GetContainerLogs,
+			GetPodsFromKubernetes: getPodsFromKubernetes,
+			GetPods:               p.GetPods,
+			StreamIdleTimeout:     cfg.StreamIdleTimeout,
+			StreamCreationTimeout: cfg.StreamCreationTimeout,
+		}
+
+		api.AttachPodRoutes(podRoutes, mux, true)
+
+		s := &http.Server{
+			Handler:   mux,
+			TLSConfig: tlsCfg,
+		}
+		go serveHTTP(ctx, s, l, "pods")
+		closers = append(closers, s)
+	}
+
+	if cfg.MetricsAddr == "" {
+		log.G(ctx).Info("pod metrics server not setup due to empty metrics address")
+	} else {
```

Comment:
no need for else you can just return here, and the dedent the rest of the funciton

Reply:
This is maybe the file I didn't spend time on. Will do!

Reply:
Actually, this needs to go away so we don't import k8s.io/kubernetes.

Reply:
Done but let comments to make it easier to revert when VK addresses this import.